### PR TITLE
Added support of multi-page highlights

### PR DESCRIPTION
**Pull request by jonathanbyrne**
Also, added a launch config to debug inside VSCode because it isn't very obvious how to do that.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -16,11 +20,13 @@ export interface Scaled {
...
+
+  pageNumber: number;
 }
 
 export interface Position {
-  boundingRect: LTWH;
-  rects: Array<LTWH>;
+  boundingRect: LTWHP;
+  rects: Array<LTWHP>;
   pageNumber: number;
```
Comment (agentcooper) [inline src/types.ts:30]: Do we still need `pageNumber` here if all the rects have `pageNumber`?

**Comment by agentcooper** [comment]
Hey, this is pretty good!

I did some testing, in all cases I've highlighted from the end of page 3 `Fast type checking. The key to Flow’s speed is modularity...` till the beginning of page 4 `...good software engineering practice anyway.`

1. It seems that the scroll to position does not work right for multi-page highlights. With the highlight mentioned above, when I click the sidebar note, the browser jumps to the beginning of page 3, but the highlight is not visible. I can reproduce this in both Brave (Chrome based) and Safari.

2. In Safari the whole next page is highlighted. I wonder if there is some quick fix that can be applied to mitigate that?
image

**Comment by jonathanbyrne** [comment]
> Hey, this is pretty good!

Thank you!

> 
> I did some testing, in all cases I've highlighted from the end of page 3 `Fast type checking. The key to Flow’s speed is modularity...` till the beginning of page 4 `...good software engineering practice anyway.`
> 
> 1. It seems that the scroll to position does not work right for multi-page highlights. With the highlight mentioned above, when I click the sidebar note, the browser jumps to the beginning of page 3, but the highlight is not visible. I can reproduce this in both Brave (Chrome based) and Safari.

I will get that fixed.

> 2. In Safari the whole next page is highlighted. I wonder if there is some quick fix that can be applied to mitigate that?
> 
> image

I will fix this too.  I forgot all about testing in Safari.

> It would also be great if you can add a test about the multi-page highlight to the `./tests/integration.test.js`.

🤦‍♂️  I am sorry about that.  I had every intention of writing a test for my change but completely forgot about it at the end.

File: src/types.ts (hunk 1 above)
Comment (jonathanbyrne) [reply src/types.ts:30]: I believe you are correct.  I will check that out.

Comment (jonathanbyrne) [reply src/types.ts:30]: I looked at this further and my thought is that `pageNumber` should probably stay for the purpose of backwards compatibility.  If it is removed, older versions of the highlights would not display correctly.  We store the highlights in a database and display them when a PDF is loaded.  I suspect a lot of other people do the same.

**Comment by jonathanbyrne** [comment]
> > 1. It seems that the scroll to position does not work right for multi-page highlights. With the highlight mentioned above, when I click the sidebar note, the browser jumps to the beginning of page 3, but the highlight is not visible. I can reproduce this in both Brave (Chrome based) and Safari.
> 
> I will get that fixed.

This is now fixed.

> > 1. In Safari the whole next page is highlighted. I wonder if there is some quick fix that can be applied to mitigate that?
> 
> I will fix this too. I forgot all about testing in Safari.
> 

I am 95% sure I found the problem and fixed it.  However, I do not have a Mac to test on, so I cannot be 100% certain.

> > It would also be great if you can add a test about the multi-page highlight to the `./tests/integration.test.js`.
> 
> 🤦‍♂️ I am sorry about that. I had every intention of writing a test for my change but completely forgot about it at the end.

It's not the greatest test in the world, but I think it confirms that multipage selects work.  This was my first time using Puppeteer and I couldn't debug the test while the browser was visible so I was flying blind.  I'm running Ubuntu on WSL so things can get a bit goofy.

**Comment by jonathanbyrne** [comment]
Sorry about those build errors.  It should be all fixed now.

**Comment by agentcooper** [comment]
Great work, thanks! Released in 5.3.0.

**Pull request merged**